libs/ml/tests/ml/dataloader/skipgram_loader.cpp

```diff
@@ -81,37 +72,20 @@ TYPED_TEST(SkipGramDataloaderTest, loader_test)
        std::pair<std::string, std::string>("total", "of"),
        std::pair<std::string, std::string>("total", "length"),
        std::pair<std::string, std::string>("length", "total"),
-       std::pair<std::string, std::string>("length", "ten"),
-       std::pair<std::string, std::string>("ten", "length"),
-       std::pair<std::string, std::string>("ten", "words"),
-       std::pair<std::string, std::string>("words", "ten")});
+       std::pair<std::string, std::string>("length", "ten")});
 
-  std::vector<TypeParam> left_and_right;
   for (std::size_t j = 0; j < 100; ++j)
   {
     if (loader.IsDone())
     {
       loader.Reset();
     }
-    left_and_right            = loader.GetNext().second;
-    std::string input         = loader.VocabLookup(SizeType(double(left_and_right.at(0).At(0, 0))));
-    std::string context       = loader.VocabLookup(SizeType(double(left_and_right.at(1).At(0, 0))));
-    auto        input_context = std::make_pair(input, context);
+    auto        left_and_right = loader.GetNext().second;
```

Comment:
slightly pedantic point but:
use of auto here is not preferred because other developers can't immediately see what type this object is.